File: producer-java/Producer.java
import com.rabbitmq.client.*;

public class Producer {
    public static void main(String[] argv) throws Exception {
Thread.sleep(15000);
        ConnectionFactory factory = new ConnectionFactory();
        factory.setHost("rabbitmq");
        factory.setUsername("miusuario");
        factory.setPassword("mipassword");

        try (Connection connection = factory.newConnection();
             Channel channel = connection.createChannel()) {

            channel.queueDeclare("hello", false, false, false, null);

            int i = 1;
            while (true) {
                String message = "Mensaje " + i++ + " desde Java Producer 🚀";
                channel.basicPublish("", "hello", null, message.getBytes("UTF-8"));
                System.out.println("📤 Enviado: " + message);
                Thread.sleep(2000); // Espera 2 segundos
            }
        }
    }
}

File: producer-csharp/Producer.cs
using System;
using System.Text;
using System.Threading;
using RabbitMQ.Client;

namespace App
{
    class Producer
    {
        static void Main(string[] args)
        {
Thread.Sleep(15000);
            var factory = new ConnectionFactory()
            {
                HostName = "rabbitmq",
                UserName = "miusuario",
                Password = "mipassword"
            };

            using var connection = factory.CreateConnection(new string[] { "rabbitmq" });
            using var channel = connection.CreateModel();

            channel.QueueDeclare(queue: "hello",
                                 durable: false,
                                 exclusive: false,
                                 autoDelete: false,
                                 arguments: null);

            int i = 1;
            while (true)
            {
                string message = $"Mensaje {i++} desde C# Producer 🚀";
                var body = Encoding.UTF8.GetBytes(message);

                channel.BasicPublish(exchange: "",
                                     routingKey: "hello",
                                     basicProperties: null,
                                     body: body);

                Console.WriteLine($"📤 Enviado: {message}");
                Thread.Sleep(2000); // Espera 2 segundos
            }
        }
    }
}
